docs/program/_index.md

```diff
@@ -3,8 +3,278 @@ title: "Program a Robot"
+import { Client, createRobotClient, RobotClient } from '@viamrobotics/sdk';
+
+async function connect() {
+  // You can remove this block entirely if your robot is not authenticated.
+  // Otherwise, replace with an actual secret.
+  const secret = '<SECRET>';
+  const credential = {
+    payload: secret,
+    type: 'robot-location-secret',
+  };
+
+  // Replace with the host of your actual robot running Viam.
+  const host = "<HOST>";
+
+  // Replace with the signaling address. If you are running your robot on Viam,
+  // it is most likely https://app.viam.com:443.
+  const signalingAddress = 'https://app.viam.com:443';
+
+  const iceServers = [{ urls: 'stun:global.stun.twilio.com:3478' }];
+
+  return createRobotClient({
+    host,
+    credential,
+    authEntity: host,
+    signalingAddress,
+    iceServers
+  });
+}
+
+async function main() {
+  // Connect to client
+  let client: Client;
+  try {
+    client = await connect();
+    console.log('connected!');
+
+    let resources = await client.resourceNames();
+    console.log('Resources:');
+    console.log(resources);
+  } catch (error) {
+    console.log(error);
+    return;
+  }
+}
+
+main();
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+{{< alert title="Note" color="note" >}}
+The C++ SDK is currently in alpha.
+{{< /alert >}}
+
+```cpp {class="line-numbers linkable-line-numbers"}
+# include <string>
+# include <vector>
+
+# include <boost/optional.hpp>
+
+# include <viam/api/common/v1/common.pb.h>
+# include <viam/api/robot/v1/robot.grpc.pb.h>
+
+# include <viam/sdk/robot/client.hpp>
+# include <viam/sdk/components/camera/client.hpp>
+
+using namespace viam::sdk;
+
+int main() {
+  std::string host("ADDRESS FROM THE VIAM APP");
+  DialOptions dial_opts;
+  Credentials credentials("SECRET FROM THE VIAM APP");
+  dial_opts.set_credentials(credentials);
+  boost::optional<DialOptions> opts(dial_opts);
+  Options options(0, opts);
+
+  auto robot = RobotClient::at_address(host, options);
+
+  std::cout << "Resources:\n";
+  for (const ResourceName& resource: *robot->resource_names()) {
+    std::cout << resource.namespace_() << ":" << resource.type() << ":"
+              << resource.subtype() << ":" << resource.name() << "\n";
+  }
+
+  return 0;
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+{{% /expand%}}
+
+Save this file on your development machine with the file type of your preferred SDK.
+
+Edit the boilerplate code by [adding control logic](/program/sdks/) to make a client application that connects to your robot and controls it in the way you want.
+
+Then, [execute this program](/program/run/) on any computer as long as:
+
+1. You have [installed the appropriate SDK](#requirements) and language on this machine.
+2. The program establishes a connection to your robot through the cloud, on a local or wide area network (LAN or WAN), or [locally](/program/run/#run-code-locally) when [executed](/program/run/).
+
+### `import`: Find Required Imports
```

Comment:
I think this and the following section could instead be embedded in the text before you say "Edit the boilerplate" because the you explain the boilerplate first. If you want to add section headings maybe "Copy the Code Sample" and "Edit and run your code" would work better?

Reply:
thoughts?

Reply:
I think editing the boilerplate should come in this later section, the above sections should strictly be setting up and running the boilerplate as it was generated.

Reply:
@mcvella @npentrel Is this ok as is then? I don't have a strong opinion either way!

Reply:
I think Matt and I both think the ordering is weird (and I do also think the headlines are confusing - because you're not finding imports, they should already be there). How about this instead:

```
## Hello World: The Code Sample Tab

Create a program in the language of your choice that connects to your robot and uses methods built into the SDK's client API libraries to [interact with and control](/program/sdks/) the {{< glossary_tooltip term_id="resource" text="resources" >}} on the robot.

Start by navigating to your robot's page on [the Viam app](https://app.viam.com/robots).
Select the **Code Sample** tab, select your preferred SDK, and copy the code generated for you.
This code snippet imports all the necessary libraries to set up a connection with your robot and interface with its configured components and services.

Your boilerplate code sample should look similar to this:

{{%expand "Click this to see example boilerplate code from the Code Sample tab" %}}
...
{{% /expand%}}

Save this file on your development machine with the file type of your preferred SDK.

The sample code contains the required imports as well as the connect logic which establishes a connection for your client application to [communicate with](/internals/robot-to-robot-comms/) the robot's `viam-server` instance.
This section of the boilerplate code contains your robot's address and location secret.
You can think of these as keys or access tokens to your robot that are important to keep private.
This connection must be established for your program to be executed properly on your robot.

{{< readfile "/static/include/snippet/secret-share.md" >}}

### Run the sample code

Once you have saved the sample code, [execute your program](/program/run/).

You can run your program on any computer which:

1. has [the appropriate SDK](#requirements) installed
2. can establish a connection to your robot through the cloud, on a local or wide area network (LAN or WAN), or [locally](/program/run/#run-code-locally)

The program will connect to your robot and print a list of the available {{< glossary_tooltip term_id="resources" text="resources" >}}.

### Edit the sample code

Once you have successfully run the sample code, you can edit the boilerplate code by [adding control logic](/program/sdks/) to make a client application that connects to your robot and controls it in the way you want.
You can find the right libraries to import for SDK methods, typing, interfaces, and utilities at the start of [each resource's API documentation](/program/sdks/resource-apis/), as well as in the individual SDK documentation sites and [on GitHub](https://github.com/viamrobotics/rdk).
```

Reply:
Done, thanks for taking the time to type that out!